_posts/zh/newsletters/2024-03-27-newsletter.md

```diff
@@ -0,0 +1,105 @@
+
+- **<!--mempoolbased-feerate-estimation-->基于交易池的手续费估计**：Abubakar Sadiq Ismail 在 Delving Bitcoin 论坛中[提出][ismail fees]了使用来自节点本地交易池的信息来优化 Bitcoin Core 的[手续费率估计][topic fee estimation]的想法。当前，Bitcoin Core 通过记录每一笔交易到达交易池的区块高度、其被确认的区块高度、其费率，来产生估计。在所有信息都知晓时，到达时间（高度）与确认时间（高度）的差值会被用来更新一个表示一系列费率的桶（bucket）的指数加权移动平均值。举个例子，一笔交易如果经过 100 个区块才得到确认，其手续费率是 1.1 聪/vbyte，那么它会被合并到 1 聪/vbyte 的桶的平均值中。
+
+    这种方法的一个优势是抵御操纵：所有交易都必须既得到转发（从而对所有矿工开放）、又得到确认（不能违反任何共识规则）。而劣势则是，它仅在新区块到来时才更新一次，可能远远落后于其它使用实时交易池信息的估计。
+
+    Ismail 采纳了之前关于将交易池数据整合到费率估计的[讨论][bitcoin core #27995]，写了一些预备代码，还运行了一项分析，显示当前的算法与新算法的比较（不包含一些安全性检查）。对该主题的一个[回复][harding fees]也链接了之前由 Kalle Alm 做的同主题[研究][alm fees]，将讨论引向交易池信息是否应该既能调高估计值、又能调低估计值，还是应该只用于调低估计值。两者皆可的好处在于它整体上让估计变得更加有用；而只能用来调低估计值（同时仅使用当前基于确认速度的算法来调高估计值）的好处是更能低于操纵和正反馈循环。
+
+    截至本刊撰写之时，讨论还在继续。
+
+## 来自 Bitcoin Stack Exchange 的精选问答
+
+*[Bitcoin Stack Exchange][bitcoin.se] 是 Optech 贡献者们找寻疑问解答的首选之地，也是他们有闲暇时会帮助好奇和困惑用户的地方。在这个月度栏目中，我们挑出了自上次出刊以来出现的一些高票问题和答案。*
+
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
+nswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [<!--what-are-the-risks-of-running-a-presegwit-node-0121-->运行未激活隔离见证升级的节点软件（0.12.1 版）会有什么风险？]({{bse}}122211) Michael Folkson、Vojtěch Strnad 和 Murch 列举了个人用户运行 Bitcoin Core 0.12.1 版本软件的缺点，包括收到无效交易和区块的更高风险、在重复花费攻击面前更加脆弱、更依赖于其它节点来运行更新后的共识验证、更慢的区块验证速度、缺失许多性功能优化、无法使用 “[致密区块中继][topic compact block relay]”、不转发现在占比约为 95% 的未确认交易、[手续费估计][topic fee estimation]更不准确，以及，留有在过去的更新中修复的安全问题。0.12.1 版本的钱包用户也会错过围绕 [miniscript][topic miniscript]、[描述符][topic descriptors]钱包的开发，以及由[隔离见证][topic segwit]、[taproot][topic taproot] 和 [Schnorr 签名][topic schnorr signatures]带来的手续费节省和额外的脚本能耐。如果人们现在转而广泛采用 Bitcoin Core 0.12.1，对比特币网络的影响可能有：无效区块被网络接受的概率（以及相应的区块重组风险）升高、因为旁支区块增多而加重挖矿中心化趋势，运行该版本的矿工的挖矿收益也会降低。
+
+- [<!--when-is-opreturn-cheaper-than-opfalse-opif-->OP_RETURN 会比 OP_FALSE OP_IF 方案更便宜吗？]({{bse}}122321) Vojtěch Strnad 细数了基于 `OP_RETURN` 的数据嵌入方案与基于 `OP_FALSE OP_IF` 的方案的开销，结论是 “`OP_RETURN` 更加便宜，少了 143 字节”。
+
+- [<!--why-does-bip340-use-secp256k1-->为什么 BIP-340 会使用 secp256k1 曲线？]({{bse}}122268) Pieter Wuille 解释了为 [BIP340][] Schnorr 签名选择 secp256k1 曲线而不是 Ed25519 曲线的推理，并指出，“复用现有的公钥派生基础设施” 以及 “不改变安全性假设” 是其中的理由。
+
+- [<!--what-criteria-does-bitcoin-core-use-to-create-block-templates-->Bitcoin Core 使用什么条件来创建区块模板？]({{bse}}122216) Murch 解释 Bitcoin Core 当前为候选区块挑选交易的基于 “祖先交易集合费率” 的算法，还提到了[族群交易池][topic cluster mempool]相关的正在开展的工作，这些工作提供了多种多样的优化。
+
+- [<!--how-does-the-initialblockdownload-field-in-the-getblockchaininfo-rpc-work-->initialblockdownload 字段在 getblockchaininfo RPC 中是如何工作的？]({{bse}}122169) Pieter Wuille 指出，在节点启动之后，`initialblockdownload` 变成假值要满足两个条件：
+
+  1. “当前使用的链拥有至少跟软件硬编码的常量一样高的累积工作量证明”
+  2. “当前使用的链的最新区块的时间戳在本地时间的 24 小时以内”
+
+## 新版本和候选版本
+
+*热门比特币基础设施项目的新版本和候选版本。请考虑升级到新版本或帮助测试候选版本。*
+
+- [Bitcoin Core 26.1rc2][] 是这个主流全节点实现的一个维护版本的候选发布。
```

Comment:
Formatting: newline should be after this line I think

Reply:
done